uvicorn/protocols/http.py

```diff
@@ -237,86 +194,82 @@ def data_received(self, data):
         try:
             self.request_parser.feed_data(data)
         except httptools.HttpParserUpgrade:
-            websocket_upgrade(self)
-
-    # Flow control...
-    def pause_writing(self):
-        self.write_paused = True
-
-    def resume_writing(self):
-        self.write_paused = False
-
-    def check_pause_reading(self):
-        if self.transport is None or self.read_paused:
-            return
-        if (self.buffer_size > self.high_water_limit or
-            len(self.pipeline_queue) >= self.max_pipelined_requests):
-            self.transport.pause_reading()
-            self.read_paused = True
-
-    def check_resume_reading(self):
-        if self.transport is None or not self.read_paused:
-            return
-        if (self.buffer_size < self.low_water_limit and
-            len(self.pipeline_queue) < self.max_pipelined_requests):
-            self.transport.resume_reading()
-            self.read_paused = False
+            self.close()
 
     # Event hooks called back into by HttpRequestParser...
     def on_message_begin(self):
-        self.message = self.base_message.copy()
-        self.channels = self.base_channels.copy()
+        self.scope = None
         self.headers = []
+        self.body = b''
+        self.queue = None
 
     def on_url(self, url):
         parsed = httptools.parse_url(url)
         method = self.request_parser.get_method()
         http_version = self.request_parser.get_http_version()
-        self.message.update({
+        self.scope = {
+            'type': 'http.request',
```

Comment:
The connection scope type needs to be set to `'type': 'http'`. 

https://github.com/django/asgiref/blob/master/specs/www.rst#connection-scope

Reply:
Thanks!